### PR TITLE
Retain sliced-out coords from WCS in GlobalCoords

### PR DESCRIPTION
**Pull request by rosteen**
### Description

Implements half of #334 . I still need to add tests, but I'm creating the PR now to reserve the number for the changelog entry.

### CODE REVIEW

File: ndcube/global_coords/global_coords.py [hunk 1]
```diff
@@ -28,6 +30,20 @@ def _all_coords(self):
...
         """
+        if self._ndcube is None:
+            return self._internal_coords
+
+        if hasattr(self._ndcube.wcs.low_level_wcs, "dropped_world_dimensions"):
+            dropped_world = self._ndcube.wcs.low_level_wcs.dropped_world_dimensions
+            extra_internal = {}
+            if dropped_world != {}:
+                for i in range(len(dropped_world["value"])):
+                    name = "dropped_world_axis_{}".format(i)
```
Comment (Cadair) [inline ndcube/global_coords/global_coords.py:41]: I think given all the other places we use names and stuff, I think the best thing to do here would be to try the name key and fallback to the physical type.
```suggestion
                    name = dropped_world["world_axis_names"][i] or dropped_world["world_axis_physical_types"][i]
                    
```

File: ndcube/global_coords/global_coords.py [hunk 2]
```diff
@@ -28,6 +30,20 @@ def _all_coords(self):
         A dynamic dictionary of all global coordinates, stored here or derived
         from the ndcube object.
         """
+        if self._ndcube is None:
+            return self._internal_coords
+
+        if hasattr(self._ndcube.wcs.low_level_wcs, "dropped_world_dimensions"):
+            dropped_world = self._ndcube.wcs.low_level_wcs.dropped_world_dimensions
+            extra_internal = {}
```
Comment (Cadair) [inline ndcube/global_coords/global_coords.py:38]: Changing this name to be a little more descriptive would be good, maybe:
```suggestion
            wcs_dropped = {}
```

**Review by Cadair — COMMENTED** [review]
This is looking really good, thanks for doing this. Just a couple of tweaks.

File: ndcube/global_coords/global_coords.py (hunk 1 above)
Comment (rosteen) [reply ndcube/global_coords/global_coords.py]: I didn't realize this syntax worked for empty strings as well as `None`, good to know. Made the changes suggested and updated the tests accordingly.

File: ndcube/global_coords/global_coords.py [hunk 3]
```diff
@@ -28,6 +30,21 @@ def _all_coords(self):
...
+        if self._ndcube is None:
+            return self._internal_coords
+
+        if hasattr(self._ndcube.wcs.low_level_wcs, "dropped_world_dimensions"):
+            dropped_world = self._ndcube.wcs.low_level_wcs.dropped_world_dimensions
+            wcs_dropped = {}
+            if dropped_world != {}:
+                for i in range(len(dropped_world["value"])):
+                    name = dropped_world["world_axis_names"][i] or\
+                        dropped_world["world_axis_physical_types"][i]
```
Comment (Cadair) [inline ndcube/global_coords/global_coords.py:42]: I have a personal hatred of line continuation characters lol
```suggestion
                    name = (dropped_world["world_axis_names"][i] or
                            dropped_world["world_axis_physical_types"][i])
```

**Review by Cadair — APPROVED** [review]

File: ndcube/global_coords/global_coords.py (hunk 3 above)
Comment (rosteen) [reply ndcube/global_coords/global_coords.py:42]: A valid stance!

**Review by DanRyanIrish — APPROVED** [review]

**Comment by DanRyanIrish** [comment]
Thanks @rosteen!

**Pull request merged**